### PR TITLE
Refs #28924 - Drop qpid, add Artemis

### PR DESCRIPTION
**Pull request by jturel**
needs to be merged together:
- this PR
- https://github.com/theforeman/puppet-candlepin/pull/143
- https://github.com/Katello/katello/pull/8563

### CODE REVIEW

**Comment by ekohl** [comment]
Should this be `Refs #28924`? Are there preconditions to merging this?

File: manifests/qpid.pp [hunk 1]
```diff
@@ -26,20 +25,4 @@
   }
 
   contain qpid
-
-  qpid::config::queue { $katello::params::candlepin_event_queue:
```
Comment (ekohl) [inline manifests/qpid.pp]: Do we need to care about removing this queue?

**Comment by jturel** [comment]
> Should this be `Refs #28924`? Are there preconditions to merging this?

Yes that sounds right :+1:

File: manifests/qpid.pp (hunk 1 above)
Comment (jturel) [reply manifests/qpid.pp]: It won't be a problem since it will land in tandem with the artemis piece in Katello

**Comment by jturel** [comment]
re-marked as WIP until https://github.com/theforeman/puppet-candlepin/pull/143 is updated

File: manifests/qpid.pp (hunk 1 above)
Comment (ekohl) [reply manifests/qpid.pp]: But qpid will stick around for Pulp 2, will it be an issue if the queue persists there?

Comment (jturel) [reply manifests/qpid.pp]: Oh misunderstood your question. It'll just be a lonesome empty queue that no one publishes or subscribes to, so I think it's fine

File: manifests/candlepin.pp [hunk 2]
```diff
@@ -48,14 +47,6 @@
...
     consumer_system_name_pattern => '.+',
     adapter_module               => 'org.candlepin.katello.KatelloModule',
-    amq_enable                   => true,
-    amqp_keystore_password       => $certs::candlepin::keystore_password,
-    amqp_truststore_password     => $certs::candlepin::keystore_password,
-    amqp_keystore                => $certs::candlepin::amqp_keystore,
-    amqp_truststore              => $certs::candlepin::amqp_truststore,
-    qpid_hostname                => $katello::params::qpid_hostname,
-    qpid_ssl_cert                => $certs::candlepin::client_cert,
-    qpid_ssl_key                 => $certs::candlepin::client_key,
```
Comment (ehelms) [inline manifests/candlepin.pp:58]: I'll need you to add passing a new parameter:

```
artemis_client_dn => $artemis_client_dn
```

And then before the `class { 'candlepin`:` declration:

``
$artemis_client_dn = "CN=${certs::candlepin::cname}, OU=${certs::candlepin::org_unit}, O=candlepin, L=${certs::candlepin::country}, ST=${certs::candlepin::state}, C=${certs::candlepin::city}"
```

Comment (jturel) [reply manifests/candlepin.pp:58]: @ehelms  what do you think about defining `$artemis_client_dn` in certs and here we will do:

`artemis_client_dn => $certs::candlepin::artemis_client_dn` ?

Comment (ehelms) [reply manifests/candlepin.pp:58]: Done. You can now update to reference it.

**Pull request merged**